detour/mesh.go

```diff
@@ -130,6 +130,11 @@ func (m *NavMesh) SaveToFile(fn string) error {
 		return err
 	}
 
+	return m.ToWriter(f)
+}
+
+// ToWriter writes binary navigation mesh to io.Writer.
+func (m *NavMesh) ToWriter(w io.Writer) error {
```

Comment:
Good idea. It would be even better if the function was called `WriteTo` and had the following signature:

```go
WriteTo(w Writer) (n int64, err error)
```

since that would make `NavMesh` implement the [io.WriterTo](https://pkg.go.dev/io#WriterTo) interface.